cmd/attest_test.go

```diff
@@ -0,0 +1,199 @@
+package cmd
+
+import (
+	"os"
+	"testing"
+
+	"github.com/google/go-tpm-tools/client"
+	"github.com/google/go-tpm-tools/internal/test"
+)
+
+func makeOutputFile(tb testing.TB, output string) string {
+	tb.Helper()
+	file, err := os.CreateTemp("", output)
+	if err != nil {
+		tb.Fatal(err)
+	}
+	defer file.Close()
+	return file.Name()
+}
+
+func TestNonce(t * testing.T){
+	rwc := test.GetTPM(t)
+	defer client.CheckedClose(t, rwc)
+	ExternalTPM = rwc
+
+	//without nonce
+	RootCmd.SetArgs([]string{"attest","--key","AK"})
+	if err := RootCmd.Execute(); err == nil {
+		t.Error(err)
+	}
+
+	//with odd length nonce
+	RootCmd.SetArgs([]string{"attest","--nonce","12345","--key","AK"})
+	if err := RootCmd.Execute(); err == nil {
+		t.Error(err)
+	}
+
+}
+
+func TestAttestPass(t *testing.T) {
+	rwc := test.GetTPM(t)
+	defer client.CheckedClose(t, rwc)
+	ExternalTPM = rwc
+
+	tests := []struct {
+		name          string
+		key           string
+		algo          string
+		nonce         string
+		teenonce      string
+		outputText    string
+	}{
+		{"DefaultKey","","rsa","2222","",""},
+		{"Withnonce", "AK", "rsa", "2222", "", ""},
+		{"WithTextOutput", "AK", "rsa", "2222", "", "text1111"},
+		//@TODO These cases works when we are able to create gce key with ExternalTPM.
```

Comment:
Remove TODO or open an Issue and reference it.

Reply:
Yes , I have removed all TODO comments from the test file.